calc/vxlan_resolver.go

```diff
@@ -423,25 +490,80 @@ func (c *VXLANResolver) routeReady(r vxlanRoute) bool {
 		logCxt.Debug("Don't yet know the VTEP for this route")
 		return false
 	}
+
+	if _, ok := c.nodeNameToNode[c.hostname]; !ok {
+		logCxt.Debug("No cidr block yet for local node")
+		return false
+	}
+
+	if _, ok := c.nodeNameToNode[r.node]; !ok {
+		logCxt.Debug("No cidr block yet for node")
+		return false
+	}
+
 	return true
 }
 
+func (c *VXLANResolver) nodeCidr(nodeName string) (*cnet.IPNet, error) {
+	logCxt := logrus.WithField("node", nodeName)
+
+	node := c.nodeNameToNode[nodeName]
+	bgp := node.Spec.BGP
+	if bgp == nil {
+		return nil, errors.New("no bgp information for node")
+	}
+
+	_, cidr, err := cnet.ParseCIDROrIP(bgp.IPv4Address)
+	if err != nil {
+		logCxt.Error(err.Error())
+		return nil, err
+	}
+
+	return cidr, nil
+}
+
 // kickPendingRoutes loops through the provided routes to see if there are any which are now programmable,
 // and will send any that are.
 func (c *VXLANResolver) kickPendingRoutes(pendingRouteUpdates set.Set) {
 	pendingRouteUpdates.Iter(func(item interface{}) error {
 		r := item.(vxlanRoute)
-		logCxt := logrus.WithField("route", r)
+		logCxt := logrus.WithField("route", r).WithField("debug", "brian")
 		if !c.routeReady(r) {
 			return nil
 		}
 		logCxt.Info("Sending VXLAN route update")
-		c.callbacks.OnRouteUpdate(&proto.RouteUpdate{
+
+		routeUpdate := &proto.RouteUpdate{
 			Type: proto.RouteType_VXLAN,
 			Node: r.node,
 			Dst:  r.dst.String(),
 			Gw:   c.determineGatewayForRoute(r),
-		})
+		}
+
+		pool := c.vxlanPoolForRoute(r)
+		if pool.VXLANMode == encap.CrossSubnet {
+			logCxt.Debug("CrossSubnet")
+
+			localNodeCidr, err := c.nodeCidr(c.hostname)
+			if err != nil {
+				logCxt.WithError(err).Errorf("Couldn't retrieve local node cidr")
```

Comment:
I think we probably want to stop if we hit this condition. We should probably continue the iteration with the next route since the `IsNetOverlap` below might cause a problem.

Reply:
definitely, a little oversight on my part here